src/Razor/test/Microsoft.NET.Sdk.Razor.Test/Microsoft.NET.Sdk.Razor.Test.csproj

```diff
@@ -107,7 +107,7 @@ namespace Microsoft.AspNetCore.Razor.Design.IntegrationTests
 
   </Target>
 
-  <Target Name="RestoreTestProjects" BeforeTargets="Restore;Build">
+  <Target Name="RestoreTestProjects" BeforeTargets="Restore;Build" Condition="'$(DotNetBuildFromSource)' != 'true'">
```

Comment:
This is a test project. Why is it restoring or building at all when `'$(DotNetBuildFromSource)' == 'true'`?

Reply:
I don't really understand why this project exists at all either when the projects could just be restored normally. It causes a double restore. @pranavkm

This change is necessary because the "Build" target still runs on all projects in a source build, but with dummy targets that are overrriden to no-op if ExcludeFromSourceBuild is true. See https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Arcade.Sdk/tools/Empty.targets. Because this project file is using "BeforeTargets" to chain in, it too needs the exclusion to avoid restoring projects that are supposed to be ignored in a source build.

Reply:
> when the projects could just be restored normally

Just a regular project reference?

Reply:
Or added to the .sln file.